lib/packetgen/header/http/request.rb

```diff
@@ -85,6 +87,6 @@ def to_s
     end
 
     self.add_class HTTP::Request 
-    TCP.bind_header HTTP::Request, body: ->(b) { /^(CONNECT|DELETE|GET|HEAD|OPTIONS|PATCH|POST|PUT)/ =~ b }
+    TCP.bind_header HTTP::Request, body: ->(b) { /(CONNECT|DELETE|GET|HEAD|OPTIONS|PATCH|POST|PUT)/ =~ b.chars.select(&:valid_encoding?).join }
```

Comment:
Why not using HTTP::VERBS to construct the regexp? This should avoid repeating yourself.
Furthermore, the regexp may be built in same module than VERBS and may be a constant.

Reply:
This is a good point 👍  

This was left over from the initial implementation. I will try to fix this.